aict_tools/scripts/apply_disp_regressor.py

```diff
@@ -99,10 +119,28 @@ def main(configuration_path, data_path, disp_model_path, sign_model_path, chunks
         source_x = df_data[model_config.cog_x_column] + disp * np.cos(df_data[model_config.delta_column])
         source_y = df_data[model_config.cog_y_column] + disp * np.sin(df_data[model_config.delta_column])
 
-        key = config.telescope_events_key
-        append_column_to_hdf5(data_path, source_x, key, 'source_x_prediction')
-        append_column_to_hdf5(data_path, source_y, key, 'source_y_prediction')
-        append_column_to_hdf5(data_path, disp, key, 'disp_prediction')
+        if config.data_format == 'CTA':
+            for tel in df_data['tel_id'].unique():
+                table = f'/dl2/event/telescope/tel_{tel:03d}/{model_config.output_name}'
+                matching = (df_data['tel_id'] == tel)
+
+                d = df_data[['obs_id', 'event_id']].copy()
+                d['source_y_pred'] = source_y
+                d['source_x_pred'] = source_x
+                d['disp_pred'] = disp
+                d.name = model_config.output_name
+                d[matching].to_hdf(
```

Comment:
Matching a discussion in lstchain, we probably shouldn't use `to_hdf`, as it is pretty much suited only to be read and written by pandas.

Better to use `tables` directly to write the data using `tables.File.create_(c)array`

Reply:
Even with `format=table` ?
Thats disappointing